### DISCUSSION TITLE
SyntaxTree From Files Outside Compilation

### DISCUSSION THREAD

**Discussion by WaynePincence** [dotnet/roslyn#66360] [Q&A]
I have been experimenting with an Incremental Source Generator, and I was curious if there is a better way to solve this problem.

Essentially I have 2 (or more) configurations for a project, and some files are excluded.  Almost identical to how multi-targetting works in MAUI.

In my generator, the primary goal right now is to find marked methods in one configuration and generate corresponding methods in the other with a different body.  In order to discover the marked methods, I have been having to include the files in AdditionalFiles, which is not a big deal.  But, is there a better way than having to call SyntaxTree.ParseText on AdditionalFiles to dig down into the SyntaxNodes that would have been there?

I don't think there's any way to "sort-of include" the file for compilation, since we can't remove source.

I did see a [similar discussion](https://github.com/dotnet/roslyn/discussions/64692#discussion-4468992), referring to files outside the project.  My approach came to a similar solution, but I was curious if anything is better when it is in the project, but not the compilation.

**Comment by ryzngard** [comment]
> In my generator, the primary goal right now is to find marked methods in one configuration and generate corresponding methods in the other with a different body

Have you considered using https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/partial-method in a neutral location, and generating the implementation for each different configuration? 

Otherwise I think you're best bet is either doing a cli tool, which can still use Roslyn, or including them as additional files.
